crates/ruff/src/importer.rs

```diff
@@ -88,10 +71,30 @@ impl<'a> Importer<'a> {
         }
     }
 
-    /// Return the top-level [`Stmt`] that imports the given module using `StmtKind::ImportFrom`.
-    /// if it exists.
-    pub fn get_import_from(&self, module: &str) -> Option<&Stmt> {
-        self.import_from_map.get(module).copied()
+    /// Return the top-level [`Stmt`] that imports the given module using `StmtKind::ImportFrom`
+    /// preceding the given position, if any.
+    pub fn find_import_from(&self, module: &str, at: TextSize) -> Option<&Stmt> {
+        let idx = self
+            .ordered_imports
+            .partition_point(|stmt| stmt.start() < at);
```

Comment:
Hard to reason about whether this preliminary search is worth it.

Reply:
Probably not?

Reply:
If we assume that in most file all imports are at the start of the file, just `if stmt.start() > at { return false; }` in the find should be faster.